collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadList.java

```diff
@@ -665,42 +674,151 @@ public void formListDownloadingComplete(HashMap<String, FormDetails> result) {
             // Everything worked. Clear the list and add the results.
             mFormNamesAndURLs = result;
 
-            mFormList.clear();
-
-            ArrayList<String> ids = new ArrayList<String>(mFormNamesAndURLs.keySet());
-            for (int i = 0; i < result.size(); i++) {
-            	String formDetailsKey = ids.get(i);
-            	FormDetails details = mFormNamesAndURLs.get(formDetailsKey);
-                HashMap<String, String> item = new HashMap<String, String>();
-                item.put(FORMNAME, details.formName);
-                item.put(FORMID_DISPLAY,
-                		((details.formVersion == null) ? "" : (getString(R.string.version) + " " + details.formVersion + " ")) +
-                		"ID: " + details.formID );
-                item.put(FORMDETAIL_KEY, formDetailsKey);
-                item.put(FORM_ID_KEY, details.formID);
-                item.put(FORM_VERSION_KEY, details.formVersion);
-
-                // Insert the new form in alphabetical order.
-                if (mFormList.size() == 0) {
-                    mFormList.add(item);
-                } else {
-                    int j;
-                    for (j = 0; j < mFormList.size(); j++) {
-                        HashMap<String, String> compareMe = mFormList.get(j);
-                        String name = compareMe.get(FORMNAME);
-                        if (name.compareTo(mFormNamesAndURLs.get(ids.get(i)).formName) > 0) {
-                            break;
-                        }
+            HashMap<String, HashMap<String, FormDetails>> userForms =
+                    splitFormsIntoAccounts(getBaseContext(), mFormNamesAndURLs);
+            ArrayList<String> userSpinnerValues = new ArrayList<>(userForms.keySet());
+
+            //sort how forms will appear in the spinner
```

Comment:
Is there a standard for single line comments, seems like all the others starts with a space and a capitalized first letter